sympy/solvers/solveset.py

```diff
@@ -1705,6 +1706,8 @@ def linear_eq_to_matrix(equations, *symbols):
     if hasattr(symbols[0], '__iter__'):
         symbols = symbols[0]
 
+    symbols = list(uniq(symbols))
```

Comment:
Not sure `uniq` is safe to do to user input: the order must not be changed for example.

See @smichr has some code which checks the length and raises an exception: I think that is the right thing.